lwtime/nattime.go

```diff
@@ -0,0 +1,198 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2020, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// A simple relative and natural time package
+package lwtime
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type naturalAdjective string
+
+const (
+	// both plural and singular regular expressions should contain 3 capture groups
+	naturalPluralRE   = `^(last)\s(\d+)\s(years|months|weeks|days|hours|minutes|seconds)$`
+	naturalSingularRE = `^(this|current|previous|last)(\s)(year|month|week|day|hour|minute|second)$`
+
+	Today     naturalAdjective = "today"
+	Yesterday naturalAdjective = "yesterday"
+	This      naturalAdjective = "this"
+	Current   naturalAdjective = "current"
+	Previous  naturalAdjective = "previous"
+	Last      naturalAdjective = "last"
+)
+
+func (nta naturalAdjective) isValid() bool {
+	switch naturalAdjective(strings.ToLower(string(nta))) { // inline lowercase conversion
+	case Today, Yesterday, This, Current, Previous, Last:
+		return true
+	}
+	return false
+}
+
+type natural struct {
+	adjective naturalAdjective
+	num       string
+	iNum      int
+	unit      relativeUnit
+}
+
+func (nt *natural) loadRelativeUnit(u string) bool {
+	switch strings.ToLower(u) {
+	case "year", "years":
+		nt.unit = Year
+	case "month", "months":
+		nt.unit = Month
+	case "week", "weeks":
+		nt.unit = Week
+	case "day", "days":
+		nt.unit = Day
+	case "hour", "hours":
+		nt.unit = Hour
+	case "minute", "minutes":
+		nt.unit = Minute
+	case "second", "seconds":
+		nt.unit = Second
+	default:
+		return false
+	}
+	return true
+}
+
+func newNatural(s string) (natural, error) {
+	nt := natural{}
+	s = strings.ToLower(s)
+
+	// Today
+	if naturalAdjective(s) == Today {
+		nt.adjective = This
+		nt.num = "1"
+		nt.iNum = 1
+		nt.unit = relativeUnit("d")
+		return nt, nil
```

Comment:
nit: instead of reusing the `nt` above, why not just return the object this way?
```suggestion
		return natural{
			adjective: This,
			num: "1", iNum: 1,
			unit: relativeUnit("d"),
		}, nil
```

Reply:
Fixed.